### PR TITLE
log the remote IP of the host making a request

### PR DESCRIPTION
**Pull request by groob**
A redo of #1635. Depends on https://github.com/grpc/grpc-go/pull/1696 to work. If the grpc-go change is not wanted we can try another approach.

### CODE REVIEW

**Comment by groob** [comment]
With the deps changes this should be good to 🚢 .

**Comment by marpaia** [comment]
@groob did you verify that this works as intended with a launcher connected over both TLS and gRPC?

**Comment by groob** [comment]
Verified with `./build/launcher -root_directory /var/groob -hostname localhost:8080 -enroll_secret_path=/Users/victor/kolide/fleet_local_test/enroll_secret -insecure -debug` and saw

```
ts=2017-11-30T23:29:33.247981Z component=service method=SubmitDistributedQueryResults ip_addr=[::1]:55836 err=null took=4.842613ms
```

previously it was a panic.

File: server/launcher/server.go [hunk 1]
```diff
@@ -46,7 +47,9 @@ func New(
 func (hgprc *Handler) Handler(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		if r.ProtoMajor == 2 && strings.Contains(r.Header.Get("Content-Type"), "application/grpc") {
-			hgprc.ServeHTTP(w, r)
+			ctx := r.Context()
+			ctx = kithttp.PopulateRequestContext(ctx, r)
```
Comment (marpaia) [inline server/launcher/server.go]: why not

```
ctx := kithttp.PopulateRequestContext(ctx, r.Context())
```

Comment (marpaia) [inline server/launcher/server.go:51]: why not

```
ctx := kithttp.PopulateRequestContext(r.Context(), r)
```

Comment (groob) [reply server/launcher/server.go:51]: looked cleaner to me to separate it on two lines, but I don't have a strong preference here.

**Comment by marpaia** [comment]
Can you add something to the changelog as well?

**Pull request merged**